src/main/java/elements/ButtonImpl.java

```diff
@@ -0,0 +1,33 @@
+package elements;
+
+import org.openqa.selenium.WebElement;
+import util.ScrollUtils;
+import util.WaitUtils;
+
+/**
+ * Wrapper class with Button functionalities
+ */
+public class ButtonImpl extends ElementImpl implements Button {
+
+    public ButtonImpl(WebElement element) {
+        super(element);
+    }
+
+    @Override
+    public void click() {
+        try {
+            getWrappedElement().click();
+        }
+        catch (Exception ex) {
```

Comment:
I think it would be useful to narrow down exception type here

Reply:
+